torch/csrc/jit/init.cpp

```diff
@@ -155,13 +155,22 @@ void initJITBindings(PyObject* module) {
       .def(
           "_jit_pass_complete_shape_analysis",
           [](std::shared_ptr<Graph> graph, py::tuple inputs, bool with_grad) {
-            CompleteArgumentSpec spec(
-                with_grad,
-                evilDeprecatedBadCreateStackDoNotUse(inputs, graph->inputs()));
-            auto graph_inputs = graph->inputs();
-            AT_ASSERT(spec.size() == graph_inputs.size());
-            for (size_t i = 0; i < graph_inputs.size(); ++i) {
-              graph_inputs[i]->setType(spec.at(i));
+            ArgumentSpecCreator arg_spec_creator(*graph);
```

Comment:
seems like this switching from completeArgSpec to only propagate dimension tensor information rather than complete tensors. I guess shape analysis works fine in this case, but does this means that we are missing the complete tensor type information? I would imagine if the following passes need the complete information then they could not find it.

Reply:
Wouldn't the comment and code below this get back to CompleteTensorType and propagate those?

Reply:
oops you are right.. I was thinking `incompleteInferTypeFrom` does not get the complete information..